### PR TITLE
Reorganize dbld

### PR DESCRIPTION
**Pull request by szemere**
New versions of Docker (command line docker build, and DockerHub automated builds) accepts Dockerfile names as a parameter. Thus we no longer bounded to the concept of individual directory/dockerfile. With the new structure we can eliminate duplicated files. (Common distributions has very similar helper resources.)

Also changed the structure of the Dockerfiles itself.
- Separated the installation of distribution provided packages from the private repositories (obs/epel).
- Experimental introduction of pip packages (usually newer versions available, compared to the distributions stable branch)
- Common steps moved into [functions.sh](dbld/images/helpers/functions.sh)

Written a general README.md rather than the the distribution related ones. (Using Ubuntu Xenial as a sample where it is required.) - Suggestions/improvements are warmly welcome in the comments.

With the new pylint there were some - currently accepted - warning messages. (`make check` runs it)

Ubuntu Zesty (17.04.) reached EOL (2018.01.13.)
- Removed it.
- Introduced Artful Aardvark (17.10)
- Changed the default **deb, shell, tarball** targets to Xenial

Questions:
- Currently pip packages can not be verified by the `dpkg-checkbuilddeps`. (Makes sens, since they are not dpkg packages.) Should we create some "custom workaround" for addressing this case? i.e. empty dpkg packages as placeholders, or remove the already available binaries from the debian control file?
- How do you decide where to put a package? (pip, dist, obs/epel)

@czanik : I will appreciate some notes on the CentOS related part.
@mitzkia : Can you help me with the DockerHub, Travis integration?

resolves: #1822

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

File: dbld/images/helpers/functions.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+function add_czanik_repo_centos6 {
+    cd /etc/yum.repos.d
+    wget https://copr.fedorainfracloud.org/coprs/czanik/syslog-ng39epel6/repo/epel-6/czanik-syslog-ng39epel6-epel-6.repo
+}
+
+function add_czanik_repo_centos7 {
+    cd /etc/yum.repos.d
+    wget https://copr.fedorainfracloud.org/coprs/czanik/syslog-ng311/repo/epel-7/czanik-syslog-ng311-epel-7.repo
```
Comment (furiel) [inline dbld/images/helpers/functions.sh]: I'm not sure if we can do anything about this but it is unfortunate to have syslog-ng 3.11 to be hardcoded here.

File: dbld/images/helpers/functions.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+function add_obs_repo_debian {
+    DISTRO=$1
+    wget -qO - http://download.opensuse.org/repositories/home:/laszlo_budai:/syslog-ng/${DISTRO}/Release.key | apt-key add -
+    echo "deb http://download.opensuse.org/repositories/home:/laszlo_budai:/syslog-ng/"${DISTRO}" ./" | tee --append /etc/apt/sources.list.d/syslog-ng-obs.list
+    apt-get update -qq
+}
+
+function add_czanik_repo_centos6 {
+    cd /etc/yum.repos.d
+    wget https://copr.fedorainfracloud.org/coprs/czanik/syslog-ng39epel6/repo/epel-6/czanik-syslog-ng39epel6-epel-6.repo
```
Comment (furiel) [inline dbld/images/helpers/functions.sh]: I'm not sure if we can do anything about this but it is unfortunate to have syslog-ng 3.9 to be hardcoded here.

**Comment by furiel** [comment]
> Where to put the packages
I think the order should: pip, obs/epel, dist. We want to use pip because it can contain newer version of the python packages. And also we can enforce a common version. Other than that we should get anything from obs/epel if possible. If the distro version would be satisfactory, then that package should be removed from obs/epel. Anything else can come from the distribution.

File: dbld/images/required-packages/artful-obs.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+criterion-dev
```
Comment (furiel) [inline dbld/images/required-packages/artful-obs.txt]: Considering the order of package install location is: pip/obs/distro, is it possible to create common files? Do we need separate list for artful-obs, xenial-obs etc?

**Comment by furiel** [comment]
Can you add the current user to sudoers file (passwordless sudo) for sake of convenience?

**Comment by czanik** [comment]
Right now tgz generation with current git master fails:
```
Unable to find image 'balabit/syslog-ng-zesty:latest' locally
docker: Error response from daemon: manifest for balabit/syslog-ng-zesty:latest not found.
```
With this PR I can generate the tgz again.

File: dbld/images/required-packages/artful-obs.txt (hunk 3 above)
Comment (szemere) [reply dbld/images/required-packages/artful-obs.txt]: @lbudai Can you help with obs? Is it possible to give different names for the same package on different platforms? (I know from experience that it is allowed by the Ubuntu maintainers to change package names from distribution to distribution. It is possible with obs?)

**Comment by kira-syslogng** [comment]
Build FAILURE, $TEST_DESC_COMMENT

**Comment by kira-syslogng** [comment]
Build FAILURE, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by bazsi** [comment]
This looks good to me and would be great if we could merge this. Is

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build FAILURE, $TEST_DESC_COMMENT

**Comment by kira-syslogng** [comment]
Build FAILURE, $TEST_DESC_COMMENT

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build FAILURE, $TEST_DESC_COMMENT

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by bazsi** [comment]
\o/

**Pull request merged**